src/mission_companion.cpp

```diff
@@ -49,6 +65,33 @@ static const trait_id trait_NPC_CONSTRUCTION_LEV_1( "NPC_CONSTRUCTION_LEV_1" );
 static const trait_id trait_NPC_CONSTRUCTION_LEV_2( "NPC_CONSTRUCTION_LEV_2" );
 static const trait_id trait_NPC_MISSION_LEV_1( "NPC_MISSION_LEV_1" );
 
+static const int COMPANION_SORT_POINTS = 12;
+
+static const std::map<std::string, std::string> COOK_SKILL_0 =
+                                                    { {" Cook: Meat, Cooked", "meat_cooked"}, {" Cook: Fish, Cooked", "fish_cooked"},
+                                                    {" Cook: Veggy, Cooked", "veggy_cooked"}, {" Cook: Offal, Cooked", "offal_cooked"} };
+static const std::map<std::string, std::string> COOK_SKILL_1 =
+                                                    { {" Cook: Egg, Boiled", "boiled_egg"}, {" Cook: Starch", "starch"},
+                                                    {" Cook: Flatbread", "flatbread"}, {" Cook: Veggy, Cooked Wild", "veggy_wild_cooked"} };
+static const std::map<std::string, std::string> COOK_SKILL_2 =
+                                                    { {" Cook: Flour", "flour"}, {" Cook: Salt", "salt_from_salt_water"},
+                                                    {" Cook: Bread", "bread"}, {" Cook: Fruit Leather", "fruit_leather"} };
+static const std::map<std::string, std::string> COOK_SKILL_3 =
+                                                    { {" Cook: Meat Jerky", "jerky"}, {" Cook: Mushroom, Cooked", "mushroom_cooked"},
+                                                    {" Cook: Lard", "lard"}, {" Cook: Cornmeal", "cornmeal"} };
+static const std::map<std::string, std::string> COOK_SKILL_4 =
+                                                    { {" Cook: Meat Pie", "pie_meat"}, {" Cook: Meat, Smoked", "meat_smoked"},
+                                                    {" Cook: Veggy Pie", "pie_veggy"}, {" Cook: Fish, Smoked", "fish_smoked"} };
+static const std::map<std::string, std::string> COOK_SKILL_5 =
+                                                    { {" Cook: Sugar", "sugar"}, {" Cook: Mushroom, Dried", "dry_mushroom"},
+                                                    {" Cook: Fruit, Dehydrated", "dry_fruit"}, {" Cook: Sausage", "sausage"} };
+static const std::map<std::string, std::string> COOK_SKILL_6 =
+                                                    { {" Cook: Hardtack", "hardtack"}, {" Cook: Sausage, Wasteland", "sausage_wasteland"},
+                                                    {" Cook: Veggy, Pickled", "veggy_pickled_jarred"}, {" Cook: Cheese, Hard", "cheese_hard"} };
+static const std::map<std::string, std::string> COOK_SKILL_7 =
+                                                    { {" Cook: Pemmican", "pemmican"}, {" Cook: Veggy Aspic", "veggy_aspic"},
+                                                    {" Cook: Meat, Canned", "meat_canned"}, {" Cook: Meat Aspic", "meat_aspic"} };
```

Comment:
Having 7 separate maps whose names only differ by number is bad practice.
Also don't use `map` if you only want it for pairs.
Instead:
```
struct npc_cook_recipe {
    std::string description;
    itype_id result;
};
std::array<std::vector<npc_cook_recipe>, 8> recipes_at_skill;
```